compiler/ilgen/IlBuilder.cpp

```diff
@@ -1154,16 +1154,17 @@ IlBuilder::setHandlerInfo(uint32_t catchType)
    }
 
 TR::Node*
-IlBuilder::genOverflowCHKTreeTop(TR::Node *operationNode)
+IlBuilder::genOverflowCHKTreeTop(TR::Node *operationNode, bool isSignedOverflow)
    {
-   TR::Node *overflowChkNode = TR::Node::createWithRoomForOneMore(TR::OverflowCHK, 3, symRefTab()->findOrCreateOverflowCheckSymbolRef(_methodSymbol), operationNode, operationNode->getFirstChild(), operationNode->getSecondChild());
+   TR::ILOpCodes overflow = isSignedOverflow? TR::OverflowCHK: TR::UnsignedOverflowCHK;
+   TR::Node *overflowChkNode = TR::Node::createWithRoomForOneMore(overflow, 3, symRefTab()->findOrCreateOverflowCheckSymbolRef(_methodSymbol), operationNode, operationNode->getFirstChild(), operationNode->getSecondChild());
    overflowChkNode->setOverflowCHKOperation(operationNode->getOpCodeValue());
    genTreeTop(overflowChkNode);
    return overflowChkNode;
    }
 
 TR::IlValue *
-IlBuilder::operationWithOverflow(TR::ILOpCodes op, TR::Node *leftNode, TR::Node *rightNode, TR::IlBuilder **handler)
+IlBuilder::operationWithOverflow(TR::ILOpCodes op, TR::Node *leftNode, TR::Node *rightNode, TR::IlBuilder **handler, bool isSignedOverflow)
```

Comment:
if the genOverflowCHKTreeTop is going to take an overflow opcode shouldn't this take the opcode as well since the op here doesn't tell you if you want signed or unsinged overflow checking? If this is the case perhaps a better name still would be genOperationWithOverflowCHK or genArithmeticWithOverflowCHK